### ISSUE TITLE
Are there extra webpack loaders being used in some cases?

### ISSUE THREAD

**Issue by kamsar** [connectivedx/fuzzy-chainsaw#84]
I was reviewing the webpack configs and I'm not sure about the presence of some of the loaders; specifically in https://github.com/connectivedx/fuzzy-chainsaw/blob/develop/build/webpack.browser.js#L32, it's got loaders in there for jsx, json, and md (which seem like they'd be the domain of the static site generator?)

Conversely in https://github.com/connectivedx/fuzzy-chainsaw/blob/develop/build/webpack.static.js#L36, it has loaders for css, js, and images that would seem to be the domain of the browser config.

I'm not certain of any of this but I wanted to bring it up just in case it was an oversight. It confused me, anyway - but then that's webpack ;)

**Comment by kamsar** [comment]
Lending minor credence to this is that now that the webpack error handler is more sane in https://github.com/connectivedx/fuzzy-chainsaw/commit/12e90105defc5de0281ce9981f016185cb271a73, the output shows the same module being built in different child configurations, e.g. /tmp/bundle.js and the lone image ranger:

```
> fuzzy-chainsaw@1.0.0 build C:\Web\fuzzy-chainsaw
> gulp build

[23:04:05] Using gulpfile C:\Web\fuzzy-chainsaw\gulpfile.js
[23:04:05] Starting 'build'...
[23:04:05] Starting 'preClean'...
[23:04:05] Finished 'preClean' after 11 ms
[23:04:05] Starting 'buildWebpack'...

  49 tests complete (1.2s)

Child
                    [1mAsset[22m       [1mSize[22m  [1mChunks[22m  [1m[22m           [1mChunk Names[22m
        [1m[32m/assets/bundle.js[39m[22m     695 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
          [1m[32m/tmp/bundle.css[39m[22m  352 bytes       [1m0[22m  [1m[32m[emitted][39m[22m  main
    [1m[32m/assets/bundle.js.map[39m[22m     830 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
      [1m[32m/tmp/bundle.css.map[39m[22m   92 bytes       [1m0[22m  [1m[32m[emitted][39m[22m  main
    Child [1mextract-text-webpack-plugin[22m:
        
Child
                                       [1mAsset[22m      [1mSize[22m  [1mChunks[22m  [1m[22m           [1mChunk Names[22m
    [1m[32m/assets/images/randy-savage-62157432.jpg[39m[22m   22.9 kB        [1m[22m  [1m[32m[emitted][39m[22m  
                              [1m[32m/tmp/bundle.js[39m[22m    136 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
                          [1m[32m/assets/bundle.css[39m[22m   2.93 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
                          [1m[32m/tmp/bundle.js.map[39m[22m    164 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
                      [1m[32m/assets/bundle.css.map[39m[22m  95 bytes       [1m0[22m  [1m[32m[emitted][39m[22m  main
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
Child
                                            [1mAsset[22m     [1mSize[22m  [1mChunks[22m  [1m[22m           [1mChunk Names[22m
         [1m[32m/assets/images/randy-savage-62157432.jpg[39m[22m  22.9 kB        [1m[22m  [1m[32m[emitted][39m[22m  
                                   [1m[32m/tmp/bundle.js[39m[22m  2.63 MB       [1m0[22m  [1m[32m[emitted][39m[22m  main
                               [1m[32m/tmp/bundle.js.map[39m[22m  3.22 MB       [1m0[22m  [1m[32m[emitted][39m[22m  main
                      [1m[32mstyleguide/tags/Button.html[39m[22m  9.57 kB        [1m[22m  [1m[32m[emitted][39m[22m  
                     [1m[32mstyleguide/tags/Heading.html[39m[22m  6.88 kB        [1m[22m  [1m[32m[emitted][39m[22m  
                    [1m[32mstyleguide/tags/PageRoot.html[39m[22m   3.1 kB        [1m[22m  [1m[32m[emitted][39m[22m  
                      [1m[32mstyleguide/tags/Poster.html[39m[22m  7.19 kB        [1m[22m  [1m[32m[emitted][39m[22m  
                      [1m[32mstyleguide/tags/Rhythm.html[39m[22m  10.2 kB        [1m[22m  [1m[32m[emitted][39m[22m  
                     [1m[32mstyleguide/tags/Wrapper.html[39m[22m  11.1 kB        [1m[22m  [1m[32m[emitted][39m[22m  
    [1m[32mstyleguide/components/ProductDescription.html[39m[22m  11.6 kB        [1m[22m  [1m[32m[emitted][39m[22m  
Child
                         [1mAsset[22m      [1mSize[22m  [1mChunks[22m  [1m[22m           [1mChunk Names[22m
         [1m[32m/assets/styleguide.js[39m[22m   3.25 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
        [1m[32m/assets/styleguide.css[39m[22m    4.1 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
     [1m[32m/assets/styleguide.js.map[39m[22m   5.04 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
    [1m[32m/assets/styleguide.css.map[39m[22m  99 bytes       [1m0[22m  [1m[32m[emitted][39m[22m  main
    Child [1mextract-text-webpack-plugin[22m:
        
Child
                    [1mAsset[22m      [1mSize[22m  [1mChunks[22m  [1m[22m           [1mChunk Names[22m
           [1m[32m/tmp/bundle.js[39m[22m    675 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
          [1m[32m/tmp/bundle.css[39m[22m   1.79 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
       [1m[32m/tmp/bundle.js.map[39m[22m    805 kB       [1m0[22m  [1m[32m[emitted][39m[22m  main
      [1m[32m/tmp/bundle.css.map[39m[22m  92 bytes       [1m0[22m  [1m[32m[emitted][39m[22m  main
               [1m[32mindex.html[39m[22m   1.13 kB        [1m[22m  [1m[32m[emitted][39m[22m  
    [1m[32mstyleguide/index.html[39m[22m   1.01 kB        [1m[22m  [1m[32m[emitted][39m[22m  
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
    Child [1mextract-text-webpack-plugin[22m:
        
Child
                [1mAsset[22m     [1mSize[22m  [1mChunks[22m  [1m[22m           [1mChunk Names[22m
        [1m[32m/tmp/tests.js[39m[22m  1.81 MB       [1m0[22m  [1m[32m[emitted][39m[22m  main
    [1m[32m/tmp/tests.js.map[39m[22m   2.2 MB       [1m0[22m  [1m[32m[emitted][39m[22m  main
[23:04:17] Finished 'buildWebpack' after 11 s
[23:04:17] Starting 'postClean'...
[23:04:17] Finished 'postClean' after 3.57 ms
[23:04:17] Finished 'build' after 11 s
```

**Comment by krambuhl** [comment]
Any files that are referenced by multiple bundles will get scraped currently.  Any files directed towards the `/tmp` folder are instances where the files are created by webpack but we don't care about them.  The image is scrapped more than once because multiple entry files are referencing a css file that talks about that image.  

There are definitely cruft in loaders currently.  like I mentioned elsewhere, I'd like to refactor webpack config files to all pull from a `webpack.shared.js` file. In doing that change, I will remove loaders where possible, and replace some with `null-loader`.  Basically there are loaders to handle all the different file types that it could run into, some of them aren't necessary, but we still have to handle those file types.

**Comment by krambuhl** [comment]
Okay, at this point most of these have been cleaned up by changes elsewhere. There are still some improvements to happen, those should be addressed in #87.  Closing this for now.
